File: trades-service/busEventTopicMappings.js
const { busEventTypes } = require('./busEventTypes.js');

const busEventTopicMappings = {
  [busEventTypes.BUS_EVENT_TYPE_TRADE]: "trades",
  [busEventTypes.BUS_EVENT_TYPE_POSITION_STATE]: "position_updates",
  [busEventTypes.BUS_EVENT_TYPE_SETTLE_POSITION]: "position_updates",
  [busEventTypes.BUS_EVENT_TYPE_ORDER]: "orders",
  [busEventTypes.BUS_EVENT_TYPE_EXPIRED_ORDERS]: "orders",
  [busEventTypes.BUS_EVENT_TYPE_DISTRESSED_ORDERS_CLOSED]: "orders",
  [busEventTypes.BUS_EVENT_TYPE_LIQUIDITY_PROVISION]: "orders",
  [busEventTypes.BUS_EVENT_TYPE_MARKET]: "markets",
  [busEventTypes.BUS_EVENT_TYPE_MARKET_CREATED]: "markets",
  [busEventTypes.BUS_EVENT_TYPE_MARKET_UPDATED]: "markets",
  [busEventTypes.BUS_EVENT_TYPE_SETTLE_MARKET]: "markets",
  [busEventTypes.BUS_EVENT_TYPE_MARKET_DATA]: "market_data",
  [busEventTypes.BUS_EVENT_TYPE_PROPOSAL]: "governance",
  [busEventTypes.BUS_EVENT_TYPE_VOTE]: "governance",
  [busEventTypes.BUS_EVENT_TYPE_STATE_VAR]: "governance",
  [busEventTypes.BUS_EVENT_TYPE_NETWORK_LIMITS]: "governance",
  [busEventTypes.BUS_EVENT_TYPE_NETWORK_PARAMETER]: "governance",
  [busEventTypes.BUS_EVENT_TYPE_VALIDATOR_RANKING]: "validators",
  [busEventTypes.BUS_EVENT_TYPE_VALIDATOR_SCORE]: "validators",
  [busEventTypes.BUS_EVENT_TYPE_VALIDATOR_UPDATE]: "validators",
  [busEventTypes.BUS_EVENT_TYPE_LEDGER_MOVEMENTS]: "transfers",
  [busEventTypes.BUS_EVENT_TYPE_TRANSFER]: "transfers",
  [busEventTypes.BUS_EVENT_TYPE_BEGIN_BLOCK]: "blocks",
  [busEventTypes.BUS_EVENT_TYPE_END_BLOCK]: "blocks",
  [busEventTypes.BUS_EVENT_TYPE_TIME_UPDATE]: "time_updates",
  [busEventTypes.BUS_EVENT_TYPE_DISTRESSED_ORDERS_CLOSED]: "liquidations",
  [busEventTypes.BUS_EVENT_TYPE_SETTLE_DISTRESSED]: "liquidations",
  [busEventTypes.BUS_EVENT_TYPE_WITHDRAWAL]: "bridge",
  [busEventTypes.BUS_EVENT_TYPE_DEPOSIT]: "bridge"
}

const topicBusEventMappings = {
  "trades": [ busEventTypes.BUS_EVENT_TYPE_TRADE ],
  "position_updates": [ busEventTypes.BUS_EVENT_TYPE_POSITION_STATE, busEventTypes.BUS_EVENT_TYPE_SETTLE_POSITION ],
  "orders": [ busEventTypes.BUS_EVENT_TYPE_ORDER, busEventTypes.BUS_EVENT_TYPE_EXPIRED_ORDERS, busEventTypes.BUS_EVENT_TYPE_DISTRESSED_ORDERS_CLOSED, busEventTypes.BUS_EVENT_TYPE_LIQUIDITY_PROVISION ],
  "markets": [ busEventTypes.BUS_EVENT_TYPE_MARKET, busEventTypes.BUS_EVENT_TYPE_MARKET_CREATED, busEventTypes.BUS_EVENT_TYPE_MARKET_UPDATED, busEventTypes.BUS_EVENT_TYPE_SETTLE_MARKET ],
  "market_data": [ busEventTypes.BUS_EVENT_TYPE_MARKET_DATA ],
  "governance": [ busEventTypes.BUS_EVENT_TYPE_PROPOSAL, busEventTypes.BUS_EVENT_TYPE_VOTE, busEventTypes.BUS_EVENT_TYPE_STATE_VAR, busEventTypes.BUS_EVENT_TYPE_NETWORK_LIMITS, busEventTypes.BUS_EVENT_TYPE_NETWORK_PARAMETER ],
  "validators": [ busEventTypes.BUS_EVENT_TYPE_VALIDATOR_RANKING, busEventTypes.BUS_EVENT_TYPE_VALIDATOR_SCORE, busEventTypes.BUS_EVENT_TYPE_VALIDATOR_UPDATE ],
  "transfers": [ busEventTypes.BUS_EVENT_TYPE_LEDGER_MOVEMENTS, busEventTypes.BUS_EVENT_TYPE_TRANSFER ],
  "blocks": [ busEventTypes.BUS_EVENT_TYPE_BEGIN_BLOCK, busEventTypes.BUS_EVENT_TYPE_END_BLOCK ],
  "time_updates": [ busEventTypes.BUS_EVENT_TYPE_TIME_UPDATE ],
  "liquidations": [ busEventTypes.BUS_EVENT_TYPE_DISTRESSED_ORDERS_CLOSED, busEventTypes.BUS_EVENT_TYPE_SETTLE_DISTRESSED ],
  "deposits_withdrawals": [ busEventTypes.BUS_EVENT_TYPE_WITHDRAWAL, busEventTypes.BUS_EVENT_TYPE_DEPOSIT ]
}

module.exports = {
  busEventTopicMappings,
  topicBusEventMappings
}

File: broker-service-go/busEventTopicMappings.go
package main

type EventTopicType uint8

const (
	Topic_Trades EventTopicType = iota
	Topic_Orders
	Topic_PositionUpdates
	Topic_Accounts
	Topic_Assets
	Topic_Markets
	Topic_MarketData
	Topic_LedgerMovements
	Topic_Transfers
	Topic_Validators
	Topic_Blocks
	Topic_TimeUpdates
	Topic_Liquidations
	Topic_StakeLinkings
)

func GetBusEventTopicMap() map[string]string {

	return map[string]string{
		"BUS_EVENT_TYPE_TRADE": "trades",
		// "BUS_EVENT_TYPE_POSITION_STATE":           "position_updates",
		// "BUS_EVENT_TYPE_SETTLE_POSITION":          "position_updates",
		"BUS_EVENT_TYPE_ORDER":                    "orders",
		"BUS_EVENT_TYPE_EXPIRED_ORDERS":           "orders",
		"BUS_EVENT_TYPE_DISTRESSED_ORDERS_CLOSED": "orders",
		"BUS_EVENT_TYPE_LIQUIDITY_PROVISION":      "orders",
		"BUS_EVENT_TYPE_SNAPSHOT_TAKEN":           "orders", // Send snapshot events here so we can generate our own.
		"BUS_EVENT_TYPE_MARKET":                   "markets",
		"BUS_EVENT_TYPE_MARKET_CREATED":           "markets",
		"BUS_EVENT_TYPE_MARKET_UPDATED":           "markets",
		"BUS_EVENT_TYPE_SETTLE_MARKET":            "markets",
		"BUS_EVENT_TYPE_MARKET_DATA":              "market_data",
		// "BUS_EVENT_TYPE_PROPOSAL":                 "governance",
		// "BUS_EVENT_TYPE_VOTE":                     "governance",
		// "BUS_EVENT_TYPE_STATE_VAR":                "governance",
		// "BUS_EVENT_TYPE_NETWORK_LIMITS":           "governance",
		// "BUS_EVENT_TYPE_NETWORK_PARAMETER":        "governance",
		// "BUS_EVENT_TYPE_VALIDATOR_RANKING":        "validators",
		// "BUS_EVENT_TYPE_VALIDATOR_SCORE":          "validators",
		// "BUS_EVENT_TYPE_VALIDATOR_UPDATE":         "validators",
		"BUS_EVENT_TYPE_LEDGER_MOVEMENTS": "ledger_movements",
		// "BUS_EVENT_TYPE_TRANSFER":                 "transfers",
		// "BUS_EVENT_TYPE_BEGIN_BLOCK":              "blocks",
		// "BUS_EVENT_TYPE_END_BLOCK":                "blocks",
		// "BUS_EVENT_TYPE_TIME_UPDATE":              "time_updates",
		// "BUS_EVENT_TYPE_SETTLE_DISTRESSED":        "liquidations",
		// "BUS_EVENT_TYPE_WITHDRAWAL":               "deposits_withdrawals",
		// "BUS_EVENT_TYPE_DEPOSIT":                  "deposits_withdrawals",
		// "BUS_EVENT_TYPE_ACCOUNT":                  "accounts",
		"BUS_EVENT_TYPE_ASSET":         "assets",
		"BUS_EVENT_TYPE_STAKE_LINKING": "stake_linkings",
	}

}
